api/src/rpc/ext.rs

```diff
@@ -0,0 +1,51 @@
+//! Contains several extensive traits for core models;
+
+use mongodb::bson::Uuid;
+use sg_core::models::Task;
+
+use crate::{map, rpc::model::AddTask};
+
+pub trait TaskExt: Sized {
+    fn new_youtube(channel_id: impl Into<String>, parent: Uuid) -> Self;
+    fn new_bilibili(uid: impl Into<String>, parent: Uuid) -> Self;
+    fn new_twitter(id: impl Into<String>, parent: Uuid) -> Self;
+}
```

Comment:
ummm I'm thinking maybe moving this to core is a good idea (

Reply:
> ummm I'm thinking maybe moving this to core is a good idea (

Yea sure.

Reply:
https://github.com/suisei-cn/stargazer-reborn/pull/71